arctic/multi_index.py

```diff
@@ -90,6 +91,9 @@ def groupby_asof(df, as_of=None, dt_col='sample_dt', asof_col='observed_dt'):
     asof_col: ``str`` or ``int``
         Name or index of the column in the MultiIndex that is the observed date
     '''
+    if as_of:
+        if as_of.tzinfo is None and df.index.get_level_values(asof_col).tz is not None:
+            as_of = as_of.replace(tzinfo=pytz.utc)
```

Comment:
would something like this be better to align tzones ?
```
df_idx_tz = df.index.get_level_values(asof_col).tz
if as_of.tzinfo is None and df_idx_tz is not None:
   as_of = as_of.replace(tzinfo=df_idx_tz)
```

or 

```
df_idx_tz = df.index.get_level_values(asof_col).tz
if as_of.tzinfo is None and df_idx_tz is not None:
   as_of = as_of.replace(tzinfo=pytz.utc).astimezone(df_idx_tz)
```

Reply:
We generally assume that naive datetimes are 'local' don't we?

Reply:
good point - other code does just convert/force things to local tz, so I've mirrored that.